File: domain/src/main/java/pl/mobly/domain/LoadInstalledAppsUseCase.java
package pl.mobly.domain;



import java.util.List;

import io.reactivex.Observable;

public class LoadInstalledAppsUseCase extends UseCase<List<DomainApplicationInfo>, Void> {
    private SystemApi systemApi;

    public LoadInstalledAppsUseCase(SystemApi systemApi) {
        this.systemApi = systemApi;
    }

    @Override
    public Observable<List<DomainApplicationInfo>> buildUseCase(Void params) {
        return systemApi.loadAppsList();
    }
}

File: domain/src/main/java/pl/mobly/domain/LoadInstalledAppsUseCase.kt
package pl.mobly.domain


import io.reactivex.Observable

class LoadInstalledAppsUseCase(private val systemApi: SystemApi) : NoParamUseCase<List<DomainApplicationInfo>> {

	override fun buildUseCase(): Observable<List<DomainApplicationInfo>> {
		return systemApi.loadAppsList()
	}
}
